libs/clj-icat-direct/src/clj_icat_direct/queries.clj

```diff
@@ -565,28 +560,28 @@
            MAX(a.access_type_id)                  as access_type_id
       FROM r_coll_main c
       JOIN r_objt_access a ON c.coll_id = a.object_id
-      JOIN parent p ON c.parent_coll_name = p.coll_name
       JOIN r_objt_metamap mm ON mm.object_id = c.coll_id
       JOIN r_meta_main m ON m.meta_id = mm.meta_id 
      WHERE a.user_id IN ( SELECT group_user_id FROM user_groups )
        AND c.coll_type != 'linkPoint'
+       AND c.parent_coll_name = ?
        AND m.meta_attr_name = 'ipc_UUID'
   GROUP BY dir_name, full_path, base_name, c.create_ts, c.modify_ts, type, data_size, uuid
   ORDER BY base_name ASC"
 
    :count-files-in-folder
-   "WITH user_groups AS ( SELECT g.*
+   "WITH user_groups AS ( SELECT g.group_user_id
                             FROM r_user_main u
                             JOIN r_user_group g ON g.user_id = u.user_id
                            WHERE u.user_name = ?
                              AND u.zone_name = ? ),
 
-         parent      AS ( SELECT * from r_coll_main
+         parent      AS ( SELECT coll_id from r_coll_main
                            WHERE coll_name = ? ),
 
-         data_objs   AS ( SELECT *
+         data_objs   AS ( SELECT data_id
                             FROM r_data_main
-                           WHERE coll_id = ANY(ARRAY( SELECT coll_id FROM parent )))
+                           WHERE coll_id IN ( SELECT coll_id FROM parent ))
```

Comment:
Why change `= ANY(ARRAY(...))` to `IN(...)` here but not above in the `count-all-items-under-folder` query?
Also, I was under the impression from @tedgin that `= ANY(ARRAY(...))` would be faster in these queries than `IN(...)`